test/dns-server.test.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  createHostedZone,
+  resetHostedZone,
+  createRecord,
+  updateRecord,
+  deleteRecord,
+  listRecords,
+  getChangeStatus,
+  RecordType,
+} from "../app/lib/dns-server";
+
+describe("DNS server lib function test", () => {
+  let hostedZoneId: string;
+
+  beforeAll(async () => {
+    hostedZoneId = await createHostedZone("starchart.com");
+  });
+
+  afterAll(() => resetHostedZone(hostedZoneId));
+
+  test("Hosted zone is created and hosted zone is returned", () => {
+    expect(hostedZoneId.length).toBeGreaterThan(0);
+    expect(/^[A-Z0-9]+/.test(hostedZoneId)).toBe(true);
+  });
+
+  test("Create a record in existing hosted zone but throw an error in non existing hosted zone", async () => {
+    const changeId = await createRecord(
+      RecordType.A,
+      "sub.starchart.com",
+      "192.168.0.1",
+      hostedZoneId
+    );
+
+    expect(changeId).toMatch(/\/change\/[A-Z0-9]+/);
```

Comment:
According to the [docs](https://docs.aws.amazon.com/Route53/latest/APIReference/API_GetChange.html#API_GetChange_RequestSyntax), I think we should just check that it's a string greater than 1.

Reply:
Will be updated